### PR TITLE
Initial commit on related work revisions.

### PR DESCRIPTION
**Pull request by sabbyanandan**
Includes:   
- New addition to Storm comparison
- New section on Samza comparison

### CODE REVIEW

File: spring-xd-related-work.tex [hunk 1]
```diff
@@ -49,15 +49,65 @@ \subsection{Spring XD and Storm}
 \item Building upon the functional stream processing model, users have the option
 to choose from Reactor\cite{reactor}, Spark Streaming or RxJava APIs to build
 complex data centric applications.
+\item As an alternative to linear streaming data pipelines, Spring XD provides 
+support for directed ayclic graphs (DAG)\cite{dag} through named channels to
```
Comment (pperalta) [inline spring-xd-related-work.tex]: One of the defining characteristics of DAG is ["that there is no way to start at some vertex v and follow a sequence of edges that eventually loops back to v again."](http://en.wikipedia.org/wiki/Directed_acyclic_graph) I don't think we can make this claim with named channels. Maybe we can claim that we can do something similar, but I would not characterize it as DAG support.

Comment (sabbyanandan) [inline spring-xd-related-work.tex]: How about:

"\item As an alternative to linear streaming data pipelines, Spring XD provides 
support for directed graphs through named channels to combine multiple flows 
upstream or downstream from the message channel. The behavior of the channel 
can be either queue-based on topic-based. However, Storm accepts Directed 
Acyclic Graph (DAG)\cite{dag} of operations that represent streaming 
computations.\*"

Comment (mbogoevici) [inline spring-xd-related-work.tex]: I think that a critical difference is that Storm's deployment unit is the Topology/DAG, whereas in our case it is the linear stream that is hinted at the end of the paragraph, but maybe it needs to be better outlined.

File: spring-xd-related-work.tex [hunk 2]
```diff
@@ -49,15 +49,65 @@ \subsection{Spring XD and Storm}
...
 
+\subsection{Samza}
+Apache Samza\cite{samza} is a distributed stream processing framework that uses 
+Kafka for messaging and YARN for fault-tolerance and other non-functional
+requirements.
+
+The following features differentiate Spring XD and Samza.
+
+\begin{itemize*}
+\item Spring XD provides out of the box adapters to read and write data fragments,
```
Comment (mbogoevici) [inline spring-xd-related-work.tex]: Not sure about 'data fragments'. We could simply say

"Spring XD provides OOTB sources and sinks for a wide variety of enterprise systems (do we reference them somewhere). In the case of Samza, the main integration point is reading and writing to Kafka topics, which requires the end user to provide a solution for integrating with other types of systems.

File: spring-xd-related-work.tex [hunk 3]
```diff
@@ -49,15 +49,65 @@ \subsection{Spring XD and Storm}
...
+unless the entire dataset needs processed as a whole. The bi-directional 
+communication between stream and batch processing in Spring XD, allows construction
+of complex chains of data processing.
+\item Spring XD's runtime abstraction allows extensions and additional
+customizations. Though extensible, the implementation in Samza should comply with 
+Samza-compatible streaming system, which generally means it is very much Kafka
+or similar.
+\item Spring XD provides built-in fault-tolerance, security, and logging; whereas, 
+Samza relies on YARN to support non-functional requirements.
+\item Spring XD's YARN support provides cluster/group semantics, which allows
```
Comment (mbogoevici) [inline spring-xd-related-work.tex:103]: Does this read like we are supporting dynamic scaling with YARN? Can you detail?

Comment (sabbyanandan) [inline spring-xd-related-work.tex:103]: Per our discussion with Janne in the past: We have a container cluster/group concept which can have different configurations (admin vs containers). We can ramp up/down these groups. We can create these groups dynamically as well. 

Samza doesn't have the concept of rebalancing when you ramp up/down. The container count in their case is static when the application starts.

File: spring-xd-related-work.tex (hunk 2 above)
Comment (sabbyanandan) [inline spring-xd-related-work.tex]: Sure thing. 

How about:
"\item Spring XD provides out of the box adapters to read and write to wide variety of enterprise systems, while Samza requires implementing the programming API, which usually is bound to 
reading and writing to Kafka topics."

File: spring-xd-related-work.tex (hunk 1 above)
Comment (sabbyanandan) [inline spring-xd-related-work.tex]: Would you agree the use of named channels could help building non-linear streams in Spring XD? I was trying to articulate just that, but of course I'm open to rephrasing it.

**Comment by mbogoevici** [comment]
Merged squashed rebased https://github.com/pperalta/spring-xd-debs-2015/commit/3195299d8efc85bc0f29921910f54e6787cf7ba7

**Pull request closed without merging**